core/tests/protractor_utils/waitFor.js

```diff
@@ -123,12 +124,16 @@ var elementAttributeToBe = async function(
 * Wait for new tab is opened
 */
 var newTabToBeCreated = async function(errorMessage, urlToMatch) {
+  var handles = await browser.driver.getAllWindowHandles();
+  await browser.waitForAngularEnabled(false);
+  await browser.switchTo().window(await handles.pop());
+  // This will wait for the page to load completely and then it will fetch
+  // the url.
+  /* eslint-disable-next-line oppia/protractor-practices */
+  await browser.sleep(10000);
```

Comment:
Is there an alternative to using `browser.sleep()`?

Reply:
I don't think so. Any thoughts on this @U8NWXD ?